tests/test_normalize_fields.py

```diff
@@ -0,0 +1,125 @@
+from concurrent.futures import ThreadPoolExecutor
+import unittest
+from datetime import datetime, timezone
+
+import preql
+
+from data_diff.database import BigQuery, MySQL, Snowflake, connect_to_uri, Oracle, DEFAULT_PRECISION
+from data_diff.sql import Select
+from data_diff import database as db
+
+from .common import CONN_STRINGS
+
+
+DATE_TYPES = {
+    db.Postgres: ["timestamp({p}) with time zone", "timestamp({p}) without time zone"],
+    db.MySQL: ["datetime({p})", "timestamp({p})"],
+    db.Snowflake: ["timestamp({p})", "timestamp_tz({p})", "timestamp_ntz({p})"],
+    db.BigQuery: ["timestamp", "datetime"],
+    db.Redshift: ["timestamp", "timestamp with time zone"],
+    db.Oracle: ["timestamp({p}) with time zone", "timestamp({p}) with local time zone"],
+    # db.Presto: ["timestamp", "timestamp with zone"],
+}
+
+
+class TestNormalize(unittest.TestCase):
+    i = 0
+
+    def _new_table(self, name):
+        self.i += 1
+        return f"t_{self.i}"
+
+    def _test_dates_for_db(self, item, precision=6):
+        db_id, conn_string = item
+
+        print(f"Testing {db_id}")
+
+        sample_date1 = datetime(2022, 6, 3, 12, 24, 35, 69296, tzinfo=timezone.utc)
+        sample_date2 = datetime(2021, 5, 2, 11, 23, 34, 500001, tzinfo=timezone.utc)
+        sample_date3 = datetime(2021, 5, 2, 11, 23, 34, 000000, tzinfo=timezone.utc)
+
+        dates = [sample_date1, sample_date2, sample_date3]
+        if db_id in (BigQuery, Oracle):
+            # TODO BigQuery doesn't seem to support timezone for datetime
+            dates = [d.replace(tzinfo=None) for d in dates]
+
+        pql = preql.Preql(conn_string)
+
+        date_types = [t.format(p=precision) for t in DATE_TYPES[db_id]]
+        date_type_tables = {dt: self._new_table(dt) for dt in date_types}
+        if db_id is BigQuery:
+            date_type_tables = {dt: f"data_diff.{name}" for dt, name in date_type_tables.items()}
+        elif db_id is MySQL:
+            pql.run_statement("SET @@session.time_zone='+00:00'")
```

Comment:
Why do you need this?

Reply:
Because mysql might have a timezone != 0 by default, and it will mess up the tests.